sonar-helm-for-iac/build.gradle.kts

```diff
@@ -15,17 +19,36 @@ val goBinariesJar by tasks.registering(Jar::class) {
 }
 artifacts.add(goBinaries.name, goBinariesJar)
 
+protobuf {
+    protoc {
+        artifact = "com.google.protobuf:protoc:${libs.versions.google.protobuf.asProvider().get()}"
+    }
+}
+
 val isCi: Boolean = System.getenv("CI")?.equals("true") ?: false
 
+// Access the protoc executable located by the protobuf plugin
+val protobuf = extensions.getByType(ProtobufExtension::class)
+val toolsField = ProtobufExtension::class.java.getDeclaredField("tools").also { it.isAccessible = true }
+val tools = toolsField.get(extensions.getByType(ProtobufExtension::class)) as ToolsLocator
+val artifactFilesField = ExecutableLocator::class.java.getDeclaredField("artifactFiles").also { it.isAccessible = true }
+val protocPathProvider = providers.provider { artifactFilesField.get(tools.protoc) as FileCollection }.map {
+    // As long as we don't use any protobuf plugins, the first file should be a discovered protoc executable.
+    it.files.first()
+}.map {
+    if (!it.canExecute()) it.setExecutable(true)
+    it
+}
+
```

Comment:
The protobuf-gradle-plugin is used in a strange, hacky way. Is it not possible to use in the way suggested by the plugin documentation?

Reply:
It can only be used to compile protobufs into Java, while we also need to generate the Go code

Reply:
You were right, it can work with Go as well. It was not clear for me from the docs at first.